src/platform/silabs/multi-ota/OtaTlvEncryptionKeyMbed.cpp

```diff
@@ -0,0 +1,78 @@
+#include "OtaTlvEncryptionKey.h"
+#include <platform/silabs/SilabsConfig.h>
+
+#include "mbedtls/aes.h"
+
+#include <string.h>
+
+namespace chip {
+namespace DeviceLayer {
+namespace Silabs {
+namespace OtaTlvEncryptionKey {
+
+using SilabsConfig = chip::DeviceLayer::Internal::SilabsConfig;
+
+CHIP_ERROR OtaTlvEncryptionKey::Import(const uint8_t * key, size_t key_len)
+{
+    if (key_len != 16) // Ensure the key length is 128 bits (16 bytes)
+    {
+        ChipLogError(DeviceLayer, "Invalid key length: %lu", (unsigned long) key_len);
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    // Store the key in a member variable for later use
+    memcpy(mKey, key, key_len);
+    mKeyLen = key_len;
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR OtaTlvEncryptionKey::Decrypt(MutableByteSpan & block, uint32_t & mIVOffset)
+{
+    constexpr uint8_t au8Iv[] = { 0x00, 0x00, 0x00, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x00, 0x00, 0x00, 0x00 };
+
+    uint8_t iv[16];
+    uint8_t stream_block[16] = { 0 };
+    size_t nc_off            = 0;
+    uint32_t offset          = 0;
+    size_t remaining         = block.size();
+
+    memcpy(iv, au8Iv, sizeof(au8Iv));
```

Comment:
We should probably we initialize iv at declaration time
uint8_t iv[16] = { 0x00, 0x00, 0x00, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x00, 0x00, 0x00, 0x00 };

Reply:
we can but then we won't be able to get the vector into a common place, and since `iv` buffer gets updated during the process of encryption/decryption. Do you suggest a MACRO instead of constant and memcpy ?

Reply:
yeah a macro with the value would be cleaner something of the sort : 

```suggestion
#define AU8IV_INIT_VALUE 0x00, 0x00, 0x00, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x00, 0x00, 0x00, 0x00

uint8_t iv[16] = { AU8IV_INIT_VALUE };
    
```

Since this array seems to be used all of the place, having the default values in a macro will limit the chance of a copy pasting error somewhere and will facilitate the maintenance.

Reply:
sure @jepenven-silabs , will move this macro to the header so that it is available to all the subsequent files, including it.